lib/user.rb

```diff
@@ -0,0 +1,34 @@
+require 'pg'
+require 'bcrypt'
+require_relative './database_connection'
+
+class User
+  attr_reader :id,:name, :email
+
+  def initialize(id:, name:, email:)
+    @id = id
+    @name = name
+    @email = email
+  end
+
+  def self.create(name:, email:, password:)
+    encrypted_password = BCrypt::Password.create(password)
+    result = DatabaseConnection.query("INSERT INTO users (name, email, password) VALUES ($1, $2, $3) RETURNING id, name, email;", [name, email, password])
+    User.new(
+      id: result[0]['id'], 
+      name: result[0]['name'], 
+      email: result[0]['email'])
+  end
+
+  def self.find(id:)
+    return nil unless id
+    result = DatabaseConnection.query(
+      "SELECT * FROM users WHERE id = $1", [id]
+    )
+    User.new(
+      id: result[0]['id'],
+      name: result[0]['name'],
+      email: result[0]['email'],
+    )
+  end
+end
```

Comment:
If you do a SELECT * WHERE why is it then creating a new user?

Reply:
I think it is pushing the credentials of the user in the database (found by the id number on line 26 [id] into a User class instance that can then be used in the rest of the code.